cmd/submit_test.go

```diff
@@ -76,11 +76,10 @@ func TestSubmit(t *testing.T) {
 		files := mf.File["files[]"]
 		for _, fileHeader := range files {
 			file, err := fileHeader.Open()
-			defer file.Close()
-
 			if err != nil {
 				t.Fatal(err)
 			}
+			defer file.Close()
 			body, err := ioutil.ReadAll(file)
```

Comment:
I believe by calling defer after checking the error you run the risk of stopping execution without ever closing the file.

Reply:
If I understand the source code for`fileHeader.Open()` correctly, it calls `os.OpenFile()` (https://golang.org/src/os/file_unix.go?s=4493:4559#L137) behind the scenes, which will either return a file _or_ an error.

That means that if we do have an error, the file will be `nil`.

So it is safe to call `defer file.Close()` after dealing with the error.